packages/table/src/table2.tsx

```diff
@@ -1098,6 +1156,49 @@ export class Table2 extends AbstractComponent2<Table2Props, TableState, TableSna
         return this.grid;
     }
 
+    /**
+     * Renders a scroll indicator overlay on top of the table body inside the quadrant stack.
+     * This component is offset by the headers and scrollbar, and it provides the overlay which
+     * we use to render automatic scrolling indicator linear gradients.
+     *
+     * @param scrollBarWidth the calculated scroll bar width to be passed in by the quadrant stack
+     * @param columnHeaderHeight the calculated column header height to be passed in by the quadrant stack
+     * @returns A jsx element which will render a linear gradient with smooth transitions based on
+     *          state of the scroll (will not render if we are already at the top/left/right/bottom)
+     *           and the state of "scroll direction"
+     */
+    private renderScrollIndicatorOverlay = (scrollBarWidth: number, columnHeaderHeight: number) => {
+        const { scrollDirection } = this.state;
+        const handleStyle = (direction: string | undefined, compare: string) => {
```

Comment:
nit: it's not an event handler, so don't use `handle...` terminology:

```suggestion
        const getStyle = (direction: string | undefined, compare: string) => {
```